src/plasma/plasma_manager.cc

```diff
@@ -660,39 +645,75 @@ int read_object_chunk(ClientConnection *conn, PlasmaRequestBuffer *buf) {
   ssize_t r, s;
   CHECK(buf != NULL);
   /* Try to read one buf_size at a time. */
-  s = buf->data_size + buf->metadata_size - conn->cursor;
+  s = buf->data_size + buf->metadata_size - buf->cursor;
   if (s > RayConfig::instance().buf_size()) {
     s = RayConfig::instance().buf_size();
   }
-  r = read(conn->fd, buf->data + conn->cursor, s);
+  r = read(conn->fd, buf->data + buf->cursor, s);
 
   int err;
   if (r <= 0) {
-    LOG_ERROR("Read error");
+    LOG_ERROR("read_object_chunk %d %s", conn->fd, buf->object_id.hex().c_str());
     err = errno;
   } else {
-    conn->cursor += r;
-    CHECK(conn->cursor <= buf->data_size + buf->metadata_size);
+    buf->cursor += r;
+    CHECK(buf->cursor <= buf->data_size + buf->metadata_size);
     /* If the cursor is equal to the full object size, reset the cursor and
      * we're done. */
-    if (conn->cursor == buf->data_size + buf->metadata_size) {
-      ClientConnection_finish_request(conn);
+    if (buf->cursor == buf->data_size + buf->metadata_size) {
+      buf->complete = true;
     }
     err = 0;
   }
   return err;
 }
 
-void process_data_chunk(event_loop *loop,
+void ignore_data_chunk(event_loop *loop,
+                       int data_sock,
+                       void *context,
+                       int events) {
+  /* Read the object chunk. */
+  ClientConnection *conn = (ClientConnection *) context;
+  PlasmaRequestBuffer *buf = conn->ignore_buffer;
+
+  if (!buf->started) {
```

Comment:
Seems better to remove this if-block and just set `buf->started = true` when allocating the buffer in `process_data_reply`.